### PR TITLE
Adjust repo for babel removal (move JS files from src/ to lib/)

### PR DESCRIPTION
**Pull request by DeeDeeG**
### Requirements for Contributing a Bug Fix

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix an existing bug. To contribute other changes, you must use a different template. You can see all templates at <https://github.com/atom/.github/tree/master/.github/PULL_REQUEST_TEMPLATE>.
* The pull request must update the test suite to demonstrate the changed functionality. For guidance, please see <https://flight-manual.atom.io/hacking-atom/sections/writing-specs/>.
* After you create the pull request, all status checks must be pass before a maintainer reviews your contribution. For more details, please see <https://github.com/atom/.github/tree/master/CONTRIBUTING.md#pull-requests>.

### Identify the Bug

As of the second-to-last commit in #31 (https://github.com/atom/electron-link/commit/6e6ad694c70940016cb8d71d877a56f93331aa48), `babel` isn't meant to run in this repository anymore. `babel` isn't available in the `devDependencies`. However, the rest of the repository expects babel to "compile" (mostly copy-paste) the files from `src/` to `lib/`, which isn't happening anymore as of #31.

The current state of the repository causes the "build" and "prepare" scripts to error out, as they had been running `babel` to compile the source code, but it is missing from the `devDependencies`.

This means `npm install` (and `npm publish`) are broken in this repository, and CI is failing.

### Description of the Change

As mentioned already in #31, the code in this repository doesn't actually need to be compiled anymore in order to be compatible with modern Node.js.

This Pull Request simply moves the JS files from`src/` to `lib/` where this package and its consumers expect them to be... and cleans up the repository in various ways:
- Delete the `build` and `prepare` scripts in `package.json`, which previously ran babel on the `src/` files. These scripts aren't needed anymore.
- Adjust the `test` files to expect the files to be in `lib/`, not `src/`.
- Stop `.gitignore`ing the `lib` directory, as that's where all the code lives now in the repository.
- Stop `.npmignore`ing the `src` directory, as it doesn't exist anymore.
  - Off-topic for this PR, but started `.npmignore`ing appveyor.yml, because consumers of this package don't need our CI config file.

### Alternate Designs

None.

### Possible Drawbacks

This PR means `babel` isn't transforming the code anymore.

If `babel` was meaningfully changing the functionality of the code (which it shouldn't), then the code will behave slightly different now.

### Verification Process

CI passes... it would be nice to see if Atom builds and passes its tests with this change as well.

### Release Notes

N/A

### CODE REVIEW

**Comment by DeeDeeG** [comment]
pinging @aminya and @sadick254...

This PR finishes womething that was started in #31 (https://github.com/atom/electron-link/pull/31/commits/6e6ad694c70940016cb8d71d877a56f93331aa48) and makes CI passing again for this repository.

File: package.json [hunk 1]
```diff
@@ -5,9 +5,7 @@
   "main": "lib/index.js",
   "scripts": {
     "test": "mocha test/**/*.test.js --ui=tdd",
-    "tdd": "electron-mocha test/**/*.test.js --ui=tdd --renderer --interactive",
-    "build": "rimraf lib && babel src -d lib",
-    "prepare": "npm run build"
+    "tdd": "electron-mocha test/**/*.test.js --ui=tdd --renderer --interactive"
```
Comment (aminya) [inline package.json:8]: Thanks for the catch. I forgot to remove build and prepare.

That is all needed. I don't understand why the files are moved.

File: package.json [hunk 2]
```diff
@@ -5,9 +5,7 @@
   "main": "lib/index.js",
```
Comment (aminya) [inline package.json:5]:
```suggestion
  "main": "src/index.js",
```

**Review by aminya — CHANGES_REQUESTED** [review]
Thanks for the catch. 

My suggestions are what I should have done. I don't understand why the files are moved.

**Comment by DeeDeeG** [comment]
That would work.

I'm not sure if anyone uses this package and manually uses `require(electron-link/lib/[some-file])`, but it would be a breaking change for those users.

**Review by sadick254 — APPROVED** [review]
This looks great. It is better if we transfer `src` to `lib` and delete `src`. As @DeeDeeG has pointed out ` this package is used outside of Atom in various other projects`.

**Pull request merged**